dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,230 @@
+        MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        RETURN post.createdAt AS created_at
+        ORDER BY post.createdAt DESC
+        LIMIT 1
+        """
+        
+        with self.driver.session() as session:
+            result = session.run(query, forum_endpoint=forum_endpoint)
+            latest_post = result.single()
+            
+            self.driver.close()
+        
+        if latest_post:
+            return latest_post["created_at"]
+        else:
+            return None
+        
+    def combine_posts_with_cathegories(self, post_details, post_categories) -> list:
+        """
+        Combine post details with their respective categories.
+
+        :param post_details: List of dictionaries containing post details.
+        :param post_categories: List of dictionaries containing post categories.
+        :return: List of combined dictionaries.
+        """
+        combined_results = []
+        for post in post_details:
+            matched = False
+            for category in post_categories:
+                if post["post_id"] == category["post_id"]:
+                    post["category_id"] = category["category_id"]
+                    post["category_name"] = category["category_name"]
+                    matched = True
+                    break
+            if not matched:
+                post["category_id"] = None
+                post["category_name"] = None
+            combined_results.append(post)
+        return combined_results
+
+    def fetch_raw_data(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch and combine post details and categories.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of combined dictionaries containing post details and categories.
+        """
+        if created_at and comparison:
+            post_details = self.fetch_post_details(created_at, comparison)
+        else:
+            post_details = self.fetch_post_details()
+
+        post_ids = [post["post_id"] for post in post_details]
+        post_categories = self.fetch_post_categories(post_ids)
+        return self.combine_posts_with_cathegories(
+            post_details, post_categories
+        )
+
+    def extract(self, period: datetime, recompute: bool = False) -> list:
+        """
+        Extract data based on the period and recompute flag.
+
+        :param period: The datetime period to filter posts.
+        :param recompute: Flag to indicate if recompute is needed.
+        :return: List of combined dictionaries containing post details and categories.
+        """
+        data = []
+        if recompute:
+            data = self.fetch_raw_data()
+        else:
+            latest_activity_date = self.get_latest_post_created_at(
```

Comment:
In case of `recompute` being False, we should check the processed data to see what was the latest document date on MongoDB rather than checking it on Neo4j.